tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -109,6 +109,46 @@ def load_vulnerability_detector_configurations(host_manager):
 
     return configurations
 
+@pytest.fixture(scope='module', autouse=True)
+def install_npm(host_manager: HostManager):
+    """Check and install npm if not already installed"""
+
+    target_os_groups = ['macos']
+
+    for group in target_os_groups:
+        for host in host_manager.get_group_hosts(group):
+            # Check if Node and npm is installed
+            logger.info(f"Checking and installing npm on {host}")
+            node_check_command = "PATH=/usr/local/bin:$PATH && command -v node"
+            node_check_result = host_manager.get_host(host).ansible(
+                "shell",
+                node_check_command,
+                become=True,
+                become_user='vagrant',
+                check=False
+            )
+            logger.info(f"Node check result on {host}: {node_check_result}")
+            # Install node if it is not already installed.
+            if node_check_result['rc'] != 0:
+                logger.info("Installing Node.js and npm")
+                # Download Node.js package
+                download_command = "curl -o /tmp/node-v21.7.1.pkg https://nodejs.org/dist/v21.7.1/node-v21.7.1.pkg"
+                download_result = host_manager.get_host(host).ansible("shell",
+                                                                      download_command,
+                                                                      become=True,
+                                                                      become_user='vagrant',
+                                                                      check=False)
+                logger.info(f"Node.js package download result on {host}: {download_result}")
+                # Install Node.js
+                node_install_command = "sudo installer -pkg /tmp/node-v21.7.1.pkg -target /"
+                node_install_result = host_manager.get_host(host).ansible("shell",
+                                                                          node_install_command,
+                                                                          become=True,
+                                                                          become_user='vagrant',
+                                                                          check=False)
+                logger.info(f"Node.js and npm installation result on {host}: {node_install_result}")
```

Comment:
Any option to use install_package method instead?

Reply:
Modified here: https://github.com/wazuh/wazuh-qa/pull/5158/commits/220e419bd3750d0c02c7026476a853bf9c3a17d8 and checked in this comment: https://github.com/wazuh/wazuh-qa/issues/5128#issuecomment-2022576419